ml-agents-envs/mlagents/envs/brain_conversion_utils.py

```diff
@@ -0,0 +1,66 @@
+from mlagents.envs.brain import BrainInfo, BrainParameters, CameraResolution
+from mlagents.envs.base_env import BatchedStepResult, AgentGroupSpec
+import numpy as np
+from typing import List, Any
+
+
+def step_result_to_brain_info(
+    step_result: BatchedStepResult,
+    group_spec: AgentGroupSpec,
+    agent_id_prefix: int = None,
+) -> BrainInfo:
+    n_agents = step_result.n_agents()
+    vis_obs_indices = []
+    vec_obs_indices = []
+    for index, observation in enumerate(step_result.obs):
+        if len(observation.shape) == 2:
+            vec_obs_indices.append(index)
+        if len(observation.shape) == 4:
+            vis_obs_indices.append(index)
+    if len(vec_obs_indices) == 0:
+        vec_obs: List[Any] = []
+    else:
+        vec_obs = np.concatenate([step_result.obs[i] for i in vec_obs_indices], axis=1)
+    vis_obs = [step_result.obs[i] for i in vis_obs_indices]
+    mask = np.ones((n_agents, np.sum(group_spec.action_size)), dtype=np.float32)
+    if group_spec.is_action_discrete():
+        mask = np.ones(
+            (n_agents, np.sum(group_spec.discrete_action_branches)), dtype=np.float32
+        )
+        if step_result.action_mask is not None:
+            mask = 1 - np.concatenate(step_result.action_mask, axis=1)
+    if agent_id_prefix is None:
+        agent_ids = list(step_result.agent_id)
+    else:
+        agent_ids = [
+            f"${agent_id_prefix}-{ag_id}" for ag_id in list(step_result.agent_id)
```

Comment:
Do you need the `list(...)` here? That allocates a python array, but you should be able to iterate through the numpy array instead.